.evergreen/run-typescript.sh

```diff
@@ -5,14 +5,25 @@ source "${PROJECT_DIRECTORY}/.evergreen/init-nvm.sh"
 
 set -o xtrace
 
-function get_current_ts_version {
-    node -e "console.log(require('./package-lock.json').dependencies.typescript.version)"
-}
+if [ -z "$TS_CHECK" ]; then echo "TS_CHECK must be set to either COMPILE_DRIVER or CHECK_TYPES"; exit 1; fi
```

Comment:
If those are the only two valid settings for this var, can we check against that being the only permitted values here.
_edit_ ah just got to the end of the script that has the fail case, that's fine, but as a reader of this line of script I'd expect it to be checked here, but I'm fine with how it is, so just leaving this as a suggestion

Reply:
Good point, I can add a check

Is it worth keeping the guard at the end of the script?  There's nothing stopping future devs from modifying `TS_CHECK` in the middle of the script, after it has been validated here but before we reach the actual code that uses it.  That's why I have the "else" block at the end.

Reply:
True, the inspiration just comes from how set but incorrect values can sometimes hide errors. The check at the end would've caught the issue when we go to test, a check at the top prevents any of that setup code (maybe npm install) from running before we get there, both have value and if its not too much to add why not be extra defensive. `#justCIThings`

Reply:
addressed

Reply:
I left the check at the end and added validation at the start